pkg/gather/common/common.go

```diff
@@ -0,0 +1,14 @@
+package common
```

Comment:
Do we need this new package (given the fact we just declare one new type here?)?

Reply:
I'm a little bit afraid that to have this `common` package inside gather can be a little bit confusing, 'cause the other folders are gatherer folders. If we need this package, I'll definitely suggest moving it to the `pkg` root, or even reorganize the gatherers folder structures by creating some `pkg/gatherers` folder.

Reply:
Well, moving `GatheringClosure` to the `gather` package would still cause circular dependency. Maybe we can reorganize things somehow differently.

Reply:
Hmm true...any suggestions how to reorganize? I am running out of ideas....or maybe just another name for the package?

Reply:
Let me take another look into it... Normally I do have some good ideas about folder organization. :rofl:

Reply:
How about this: https://github.com/rluders/insights-operator/tree/support-of-gatherers-with-different-periods

Reply:
@rluders looks great

Reply:
done